scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
+    #                errorRevert "Received Error signal from line $1"
+    #                # revert action
+    #                >&2 contactme # This function gives contact information
+    #                exit "$?"
+    #            }
+    #
+    #            # action of functions
+    #            # 'exit SING # to the trap that catches exit SIGN'
+    #        }
+    #
+    #        func
+    #
+    #    }
+    #
+    #**************************************************
+
+    # Function must have revert procedure. In case of any problem - function quits without artifacts/changes to the system
+    # Nix is pure, and claims to be transnational, so let's show people clean transnational installation
+
+    createDest() {
+
+        trap destCreateRevert 1
+
+        destCreateRevert() {
+            # all means failed, Exit!
+            error "
+
+    Can't get root access to create $dest folder. All variants failed.
+    Exiting...
+    "
+        }
+
+        # Shell parses variable with current user, so value under sudo/su not changes.
+        # It is const, but don't move it above USER set/checks.
+        readonly cmd="mkdir -m 0755 $dest && chown $USER $dest"
+
+        # Trying to create $dest by all means
+
+        # For case user is root, or has access to create $cmd. Else -> do 'then' section
+        if sh -c "$cmd" > /dev/null 2>&1 ; then
+            print "Creating $dest directory by running $cmd using $USER rights"
+        else
+            if which sudo > /dev/null 2>&1 || type sudo > /dev/null 2>&1 ; then
+                print "Creating $dest directory by running $cmd using sudo:"
+                if sudo -n true 2> /dev/null ; then
+                    print "Checked: sudo access does not require password."
+                else
+                    print "Checked: sudo access requires to enter password:"
+                fi
+                # Using sudo to create directory
+                if ! sudo sh -c "$cmd" && print "Exiting priviliged mode." ; then
+                    >&2 notice "
+
+    Password don't match or user $USER not in sudoers.
+    "
+                fi
+            else
+                print "'sudo' seems not installed"
             fi
-            added=1
-            break
+
         fi
-    done
 
-fi
+        # If dest still not created (with user or sudo) - use su to access root
+        if [ ! -d "$dest" ] ; then
+            print "Trying to get privileges using su and root password"
+            if ! su root sh -c "$cmd" && print "Exiting priviliged mode."; then
+                exit 1 # Look into the trap
+            fi
+        fi
+    }
+
+    createStore() {
+        trap createStoreRevert 1
+        createStoreRevert() {
+            error "
+
+    Script tried to create $dest/store , but it already exists.
+    Something is wrong with the process.
+    Please gather information of your setup conditions,
+    search does BUG exists make further actions.
```

Comment:
This message is incorrect, `mkdir -p` would not fail if the dir already exists. I think you'd better just write something like

```
Could not create $dest/store - please look at the error message above and fix the issue.
```

Reply:
Yes. I thought I checked, and it returnes $? - 1, but, I checked once more - and yes, it is not.

Reply:
I think the wording here needs improvements, maybe:

> I tried to make $dest/store, but it already exists and I don't own it. This seems to suggest a bug.

Please also don't use a trap here, just `if ! mkdir ...`